### PR TITLE
Swap out custom stream lib for external libs

### PR DESCRIPTION
**Pull request by JakeChampion**
This reduces the amount of utility code in our codebase and as a bonus, each external lib has tests which cover their features.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -119,139 +122,138 @@ function getPolyfillString(options) {
...
+		if ('all' in options.features) {
+			output.add(streamFromString("\nconsole.log('"+allWarnText+"');\n"));
+		}
+	});
 
-	return options.stream ? output : new Promise(resolve => {
-		const bufs = [];
-		output.on('data', chunk => bufs.push(chunk));
-		output.on('end', () => resolve(Buffer.concat(bufs).toString('UTF-8')));
+	return options.stream ? output : streamToPromise(output).then(bufs => {
```
Comment (triblondon) [inline lib/index.js]: You could use a higher level lib here like stream-to-string: https://www.npmjs.com/package/stream-to-string

File: service/routes/api.js [hunk 2]
```diff
@@ -71,13 +74,18 @@ router.get(/^\/v2\/polyfill(\.\w+)(\.\w+)?/, (req, res) => {
...
-	outputStream.on('end', () => {
-		if (req.query.callback && req.query.callback.match(/^[\w\.]+$/)) {
-			res.write("\ntypeof "+req.query.callback+"==='function' && "+req.query.callback+"();");
+	const outputStream = mergeStream();
+
+	outputStream.add(polyfillio.getPolyfillString(params));
+
+	if (req.query.callback && req.query.callback.match(/^[\w\.]+$/)) {
+		outputStream.add(streamFromString("\ntypeof "+req.query.callback+"==='function' && "+req.query.callback+"();"));
+	}
```
Comment (triblondon) [inline service/routes/api.js:83]: I much prefer this solution to the one in the other PR - the source order still matches the order of output which makes it easier to grok.

File: test/node/lib/test_index.js [hunk 3]
```diff
@@ -138,15 +138,13 @@ describe("polyfillio", () => {
...
 		it('should support streaming output', done => {
-			const ReadableStream = require('stream').Readable;
 			const buf = [];
 			const s = polyfillio.getPolyfillString({
 				features: { default: {} },
 				uaString: 'chrome/30',
 				stream: true,
 				minify: false
 			});
-			assert.instanceOf(s, ReadableStream);
```
Comment (triblondon) [inline test/node/lib/test_index.js:149]: Shouldn't we at least assert that it implements ReadableStream?

**Review by triblondon — APPROVED** [review]
I'm sad to lose my stream module, but this makes sense and at least I know a lot more about streams now!

**Comment by JakeChampion** [comment]
Thanks for the review @triblondon, useful feedback. I've made the changes you suggested and will merge this one CI passes.

**Pull request merged**